test/run-c-api-examples.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+#
+# Copyright 2016 WebAssembly Community Group participants
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import print_function
+
+import argparse
+import os
+import subprocess
+import sys
+
+import find_exe
+
+ALL_EXAMPLES = [
+    'callback',
+    'finalize',
+    'global',
+    'hello',
+    'hostref',
+    'memory',
+    'multi',
+    'reflect',
+    'serialize',
+    'start',
+    'table',
+    'threads',
+    'trap',
+]
+
+# We don't currently yet support shared modules which is required for threads.
+SKIP_EXAMPLES = [
+    'threads',  # We don't yet support threads
+    'finalize',  # This test is really slow
+]
+
+IS_WINDOWS = sys.platform == 'win32'
+
+
+def run_test(test_exe):
+    print('Running.. %s' % test_exe)
+    proc = subprocess.Popen([test_exe], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    stdout, _ = proc.communicate()
+    if proc.returncode == 0:
+        return 0
+    print("Failed with returncode=%d" % proc.returncode)
+    print(stdout)
+    print('FAIL(%d): %s' % (proc.returncode, test_exe))
+    return 1
+
+
+def error(msg):
+    print(msg)
+    sys.exit(1)
+
+
+def check_for_missing(found):
+    # Check that all the expected examples are found
+
+    basenames = [os.path.splitext(f)[0] for f in found]
+    for e in ALL_EXAMPLES:
+        if e not in basenames:
+            error('Example binary not found: %s' % e)
```

Comment:
this will exit on the first error, might be nicer to use set subtraction in both directions so you can get a full list instead.

Reply:
Done